rave/src/main/java/com/uber/rave/Rave.java

```diff
@@ -69,6 +70,34 @@ public static synchronized Rave getInstance() {
     }
 
     /**
+     * Validate elements in a collection.
+     * @param collection the collection to validate
+     * @param <T> the type of the collection
+     * @throws RaveException if validation fails.
+     */
+    public <T> void validate(Collection<T> collection) throws RaveException {
```

Comment:
What do you think about generating "deep validation" code inline at annotation processing?
 Making a change like this isn't a no-brainer to me since it really increases the surface area of RAVE; we're no longer restricting ourselves to validating `@Validated` models.

Thoughts?